tests/extract/test_decorators.py

```diff
@@ -631,14 +631,14 @@ def test_sources_no_arguments() -> None:
     def no_args():
         return dlt.resource([1, 2], name="data")
 
-    # there is no spec if no arguments
+    # there is a spec even if no arguments
     SPEC = _SOURCES[no_args.__qualname__].SPEC
-    assert SPEC is None
```

Comment:
since the spec is now created even if there are no args, i assume updating this test is in order